url/src/quirks.rs

```diff
@@ -279,11 +279,14 @@ pub fn set_pathname(url: &mut Url, new_pathname: &str) {
             && new_pathname.starts_with('\\'))
     {
         url.set_path(new_pathname)
-    } else {
+    } else if SchemeType::from(url.scheme()).is_special() || !new_pathname.is_empty() || !url.has_host() {
         let mut path_to_set = String::from("/");
         path_to_set.push_str(new_pathname);
         url.set_path(&path_to_set)
     }
+    else {
+        url.set_path(new_pathname)
+    }
```

Comment:
Please fix the cargo fmt issues https://github.com/servo/rust-url/actions/runs/8513766904/job/23336162537?pr=919